Framework/Geometry/src/Instrument/InstrumentDefinitionParser.cpp

```diff
@@ -1319,6 +1320,148 @@ void InstrumentDefinitionParser::appendLeaf(Geometry::ICompAssembly *parent,
           "Duplicate detector ID found when adding RectangularDetector " +
           name + " in XML instrument file" + filename);
     }
+  } else if (category.compare("StructuredDetector") == 0 ||
```

Comment:
This is a lot of code to add to the existing `InstrumentDefinitionParser` method here. Would it be better (a) to split this out into a separate function, (b) Put this into it's own class or free-function to avoid adding to the existing mess.